src/components/activity-page/activity-page-content.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { PrimaryEmbeddable } from "./primary-embeddable";
+import { SecondaryEmbeddable } from "./secondary-embeddable";
+import { BottomButtons } from "./bottom-buttons";
+import { PageLayouts, EmbeddableSections } from "../../utilities/activity-utils";
+
+import './activity-page-content.scss';
+
+interface IProps {
+  isFirstActivityPage: boolean;
+  isLastActivityPage: boolean;
+  onPageChange: (page: number) => void;
+  page: any;
+  pageNumber: number;
+  totalPreviousQuestions: number;
+}
+
+interface IState {
+  scrollOffset: number;
+}
+
+export class ActivityPageContent extends React.PureComponent <IProps, IState> {
+  private divRef: HTMLDivElement | null;
+  public constructor(props: IProps) {
+    super(props);
+    this.state = {
+      scrollOffset: 0
+    };
+  }
+
+  render() {
+    const { isFirstActivityPage, isLastActivityPage, page, totalPreviousQuestions } = this.props;
+    const { scrollOffset } = this.state;
+    const fullWidth = page.layout === PageLayouts.Responsive;
+    const vertical = page.layout === PageLayouts.FullWidth;
+    const primaryFirst = page.layout === PageLayouts.FullWidth || page.layout === PageLayouts.FortySixty;
+
+    const primaryEmbeddables = page.embeddables.filter((e: any) => e.section === EmbeddableSections.Interactive);
+    const secondaryEmbeddables = page.embeddables.filter((e: any) => !e.section);
+
+    const pinOffSet = page.layout !== PageLayouts.FullWidth && secondaryEmbeddables.length ? scrollOffset : 0;
+
+    const renderPrimary = this.renderPrimaryEmbeddables(primaryEmbeddables, vertical, primaryFirst && !vertical, pinOffSet);
+    const renderSecondary = this.renderSecondaryEmbeddables(secondaryEmbeddables, totalPreviousQuestions, !primaryFirst && !vertical);
+    const [first, second] = primaryFirst
+                            ? [renderPrimary, renderSecondary]
+                            : [renderSecondary, renderPrimary];
+
+    return (
+      <div className={`page-content ${fullWidth ? "full" : ""}`} data-cy="page-content">
+        <div className="name">{page.name}</div>
+        <div className="introduction" dangerouslySetInnerHTML={{ __html: page.text}}></div>
```

Comment:
I started out using `dangerouslySetInnerHTML` in the rich text work but changed to a less dangerous approach upon code review. You should be able to crib the `renderHTML()` utility function from [this commit](https://github.com/concord-consortium/question-interactives/pull/33/commits/1b7ee9a0c101dcf67f351f7638c7342e51790915) instead. You're welcome. 😉

Reply:
This is sweet!  I was going to bring this topic up in a near-future stand-up.  I hate using dangerouslySetInnerHTML.  Seems so clunky.  This is a great solution! Fixed.

Reply:
Changed to @types/dompurify to dev dependency.